Phase #3/app/model/DataProvider/FileDataProvider.cs

```diff
@@ -0,0 +1,8 @@
+public class FileDataProvider : IDataProvider
+{
+     private FileReader FileReader;
+
+     public FileDataProvider(FileReader fileReader) => this.FileReader = fileReader;
+     public string GetGradesData() => FileReader.ReadFile(@"resources/grades.json");
+     public string GetStudentsData() => FileReader.ReadFile(@"resources/students.json");
+}
```

Comment:
Violates Open-Closed principle. Could be more general.
Recommendation: A single GetData method with file path as input parameter. 
File paths could be set by a static parameter or .config file.